docs/tutorials/10_torch_qgan.ipynb

```diff
@@ -0,0 +1,999 @@
+    "We convert data arrays into tensors and create a data loader from our training data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "training_data = torch.tensor(training_data, dtype=torch.float)\n",
+    "grid_elements = torch.tensor(grid_elements, dtype=torch.float)\n",
+    "\n",
+    "# Define the training batch size\n",
+    "batch_size = 300\n",
+    "dataloader = DataLoader(training_data, batch_size=batch_size, shuffle=True, drop_last=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "### Specify Backend\n",
+    "Next, we need to choose a backend that is used to run the quantum generator.\n",
+    "The presented method is compatible with all shot-based backends (qasm, fake hardware, real hardware) provided by Qiskit.\n",
+    "\n",
+    "First, we create a quantum instance for the training where the batch size defines the number of shots."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "backend = Aer.get_backend(\"aer_simulator\")\n",
+    "qi_training = QuantumInstance(backend, shots=batch_size)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "Then we create a quantum instance for the evaluation, we choose a higher number of shots to get better insights."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "qi_sampling = QuantumInstance(backend, shots=10000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "### Initialization of quantum generator and the respective gradient\n",
+    "Next, we define a function that creates the quantum generator from a given parameterized quantum circuit. As parameters this function also takes a quantum instance to be used for data sampling. We wrap a created quantum neural network in `TorchConnector` to make use of PyTorch training."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def create_generator(qc, quantum_instance) -> TorchConnector:\n",
+    "    circuit_qnn = CircuitQNN(\n",
+    "        qc,\n",
+    "        input_params=[],\n",
+    "        weight_params=qc.parameters,\n",
+    "        quantum_instance=quantum_instance,\n",
+    "        sampling=True,\n",
+    "        sparse=False,\n",
+    "        input_gradients=True,  # TODO: !!!!!! Check out if they are required !!!!!!!\n",
```

Comment:
for the torch connector, they are required hahaha :)

Reply:
All TODOs in the tutorial are mine. Perhaps, I was too strict that time :) I'll take care of all TODOs and streamline the code slightly. I hope :)

Reply:
> for the torch connector, they are required hahaha :)

Actually not in this case. Generator is a plain `TorchConnector`, no other layers. And there's no inputs. So, in general, gradients wrt inputs are not required and that's why I added this todo. Confirmed, they are not required.